src/main/java/de/unistuttgart/informatik/fius/icge/workbench/swing/SimPanel.java

```diff
@@ -49,34 +49,42 @@ public void mouseClicked(MouseEvent e) {}
             
             @Override
             public void mousePressed(MouseEvent e) {
-                SimPanel.this._mouseDown = true;
-                SimPanel.this.updateSettings();
-                if (SimPanel.this._s.animator != null) {
-                    SimPanel.this.handleMousePress();
+                if (e.getButton() == MouseEvent.BUTTON1) {
+                    SimPanel.this._mouseDown = true;
+                    SimPanel.this.updateSettings();
+                    if (SimPanel.this._s.animator != null) {
+                        SimPanel.this.handleMousePress();
+                    }
+                    SimPanel.this._view.update();
                 }
-                SimPanel.this._view.update();
             }
             
             @Override
             public void mouseReleased(MouseEvent e) {
-                SimPanel.this._mouseDown = false;
-                SimPanel.this.updateSettings();
-                if (SimPanel.this._s.animator != null) {
-                    SimPanel.this.handleMouseRelease();
+                if (e.getButton() == MouseEvent.BUTTON1) {
+                    SimPanel.this._mouseDown = false;
+                    SimPanel.this.updateSettings();
+                    if (SimPanel.this._s.animator != null) {
+                        SimPanel.this.handleMouseRelease();
+                    }
+                    SimPanel.this._view.update();
                 }
-                SimPanel.this._view.update();
             }
             
             @Override
             public void mouseEntered(MouseEvent e) {
-                SimPanel.this._mouseInside = true;
-                SimPanel.this._view.update();
+                if (e.getButton() == MouseEvent.BUTTON1) {
+                    SimPanel.this._mouseInside = true;
+                    SimPanel.this._view.update();
+                }

```

Comment:
This change is not needed and breaks the tracking whether the mouse is inside the `simPanel` in some cases (which is the reason why the `mouseEntered()` method was overriden).